### ISSUE TITLE
Clippy to fix warnings when they are converted to errors

### ISSUE THREAD

**Issue by psandana** [rust-lang/rust-clippy#13239]
### Description

**Description**

Let's imagine we have a linter in `clippy` throwing warnings. If we want to convert them to errors, we can execute:

```
cargo clippy --workspace --verbose --all-targets --all-features --locked -- -D warnings
```
(compiles everything)

Now, after errors (formerly warnings) are reported, we want to fix them:

```
cargo clippy --workspace --verbose --all-targets --all-features --locked --fix -- -D warnings
```
(recompiles everything!)

No errors were fixed!

This is non-intuitive. I would expect, if we can make warnings as errors, `clippy` would be able to fix them.

**Request**

Make `clippy` to fix warnings when they are converted to errors.

**Comment by Jarcho** [comment]
You don't need to convert warnings to errors for `clippy --fix` to work. The only clippy warnings which will auto-fix are those with machine applicable suggestions (generally these result in no behavior change to the code). Any lint which doesn't have such a suggestion, either because there's nothing to suggest or it just isn't implemented, will not be fixed automatically.

If you want to open as issue for specific lints which could have such a suggestion, but don't, feel free to open an issue for those lints. Preferably one issue per lint so they can be tracked. Note that not all lints can have a reasonable suggestion, e.g. `undocumented_unsafe_blocks`.

The [lints list](https://rust-lang.github.io/rust-clippy/master/) documents whether a lint has a suggestion with it. See the bottom left of each lint's documentation block.
